packages/core/core/src/services/entity-validator/validators.ts

```diff
@@ -138,18 +140,24 @@ const addStringRegexValidator = (
  */
 const addUniqueValidator = <T extends strapiUtils.yup.AnySchema>(
   validator: T,
-  { attr, model, updatedAttribute, entity }: ValidatorMetas<Attribute.Any & Attribute.UniqueOption>
+  { attr, model, updatedAttribute, entity }: ValidatorMetas<Attribute.Any & Attribute.UniqueOption>,
+  options: ValidatorOptions
 ): T => {
   if (attr.type !== 'uid' && !attr.unique) {
     return validator;
   }
 
   return validator.test('unique', 'This attribute must be unique', async (value) => {
+    const isPublish = options.isDraft === false;
+
+    // When publishing, the value will not have changed so we take the value from the entity
+    const valueToCheck = isPublish ? entity?.[updatedAttribute.name] : value;
```

Comment:
whats the difference of taking it from the entity or from the value when its published? Is the "value" empty when publishing?

Reply:
Yes the value is empty when publishing but the entity will be populated